packages/nimble-components/src/rich-text/mention-list-box/index.ts

```diff
@@ -0,0 +1,294 @@
+import { normalizeString } from '../../utilities/models/string-normalizer';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-rich-text-mention-list-box': RichTextMentionListBox;
+    }
+}
+
+/**
+ * A rich text mention list box which acts as a popup for "@mention" support in editor
+ */
+export class RichTextMentionListBox extends FoundationElement {
+    /**
+     * @internal
+     */
+    @observable
+    public region?: AnchoredRegion;
+
+    /**
+     * @internal
+     */
+    @observable
+    public activeCharacter?: string;
+
+    /**
+     * @internal
+     */
+    @observable
+    public activeMappingConfigs?: MappingConfigs;
+
+    /**
+     * @internal
+     */
+    @observable
+    public open?: boolean;
+
+    /**
+     * @internal
+     */
+    @observable
+    public readonly childItems: Element[] = [];
+
+    /**
+     * @internal
+     */
+    @observable
+    public filter?: string;
+
+    /**
+     * @internal
+     */
+    public listBox!: Listbox;
+
+    @observable
+    private anchorElement?: HTMLElement;
+
+    private hasAnySelectableOption = false;
+
+    private mentionExtensionConfig?: MentionExtensionConfiguration[];
+
+    private suggestionProps!: SuggestionProps;
+
+    private readonly intersectionObserver: IntersectionObserver = new IntersectionObserver(
+        entries => {
+            if (
+                !entries[0]?.isIntersecting
+            ) {
+                this.setOpen(false);
+            }
+        },
+        { threshold: 1.0, root: document }
+    );
+
+    /**
+     * @public
+     */
+    public async selectFirstOptionIfValidOptionExists(): Promise<void> {
+        const definedElements = [
+            this.listBox?.matches(':not(:defined)')
+                ? customElements.whenDefined(this.listBox.localName)
+                : Promise.resolve()
+        ];
+        await Promise.all(definedElements);
+        this.updateSelectableOption();
+        if (this.hasAnySelectableOption) {
+            this.listBox?.selectFirstOption();
+            this.scrollOptionIntoView();
+        }
+    }
+
+    /**
+     * @public
+     */
+    public keydownHandler(event: KeyboardEvent): boolean {
+        if (!this.open) {
+            return false;
+        }
+        switch (event.key) {
+            case keyTab:
+            case keyEnter: {
+                if (
+                    this.listBox.firstSelectedOption
+                    && this.hasAnySelectableOption
+                ) {
+                    this.activateMention({
+                        href: this.listBox.firstSelectedOption.value,
+                        displayName: this.listBox.firstSelectedOption.innerText
+                    } as MentionDetail);
+                    return true;
+                }
+                return false;
+            }
+            case keyArrowDown: {
+                if (!event.shiftKey && this.hasAnySelectableOption) {
+                    this.listBox.selectNextOption();
```

Comment:
Both `selectNextOption()` and `selectPreviousOption()` are [documented as internal](https://github.com/microsoft/fast/blob/fd65254531ac3a593b32528d81b44ebf82c8e4ab/packages/web-components/fast-foundation/src/listbox/listbox.ts#L501). Maybe instead of adding all this logic ourselves, we can call `super.keydownHandler` and then handle just tab, enter, and escape?

Reply:
Updated as per the suggestion

Reply:
Sorry, we should also use the `super.keydownHandler` to navigate on Home/End keys.